integrations: Update notifications for commented events in Jira integration.

PieterCK opened the pull request:
This PR fixes:
- Comment-related events not containing issue links.
- Outdated instructions to set up the Jira integration.

Fixes part of: [CZO](https://chat.zulip.org/#narrow/stream/9-issues/topic/Jira.20.28Cloud.29.20integration.20problems)

**Screenshots and screen captures:**
- Current commented event notification.
![image](https://github.com/user-attachments/assets/256c4d16-73b5-42fd-a17e-a994653abd01)

- Updated "comment" event notification.
![image](https://github.com/user-attachments/assets/cbf4514a-be5b-41ad-912d-532c314e3594)

- "New" Jira Site administration page. 
![image](https://github.com/user-attachments/assets/ea637805-076d-4073-b4d0-0d2a38dda19d)

- Updated Jira doc.
![image](https://github.com/user-attachments/assets/dcf2bccf-7f73-4e6b-a000-9db997785170)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

PieterCK commented:
@zulipbot add "buddy review"

@kennethnrk Please review this!

kennethnrk commented:
We should add the [Jira webhook events documentation](https://developer.atlassian.com/server/jira/platform/webhooks/) in the related documentation section, just like we do for GitHub and similar webhooks.

kennethnrk commented:
The changes fix the issue of the comment related messages not containing links, other than `comment_created_no_issue_details` (I'm assuming no issue details means we don't have the url so that's fine). 

Left one suggestion, but LGTM.
@zulipbot remove "buddy review"

sbansal1999 reviewed (commented):
@PieterCK The code looks good to me.

Since the second commit updates notifications for all types of Jira comment notifications including "edited" and "deleted" and not just "commented", I would rewrite the commit message to "Update comment related notifications in Jira integration." and accordingly the commit description as well.

nit: In the first commit message, it should be `integrations: Update...`

PieterCK commented:
Updated commit summary and description.

@sbansal1999 Thanks for the review :+1:

sbansal1999 commented:
Marking this for maintainer review. @laurynmm 

@zulipbot add "maintainer review"

PieterCK commented:
@zulipbot remove "mentor review"

laurynmm commented on zerver/webhooks/jira/doc.md:
```diff
@@ -14,8 +14,12 @@ Server versions 5.2 or greater. For older versions, you'll need our
-1. Go to your Jira **Site administration** page, and click on **Jira**.
-   In the left sidebar, scroll down, and under **Advanced**, click
+1. Go to your Jira **Site administration** page. Click on the
+   **dropdown**( <i class="fa fa-ellipsis-h" aria-hidden="true"></i> )
```
I'd call this a menu icon vs dropdown, and I wouldn't put it in bold. Also, looking at our help center documentation, we don't normally add the 'aria-hidden' attribute for these icons, so let's drop that here.

laurynmm commented on zerver/webhooks/jira/doc.md:
```diff
@@ -14,8 +14,12 @@ Server versions 5.2 or greater. For older versions, you'll need our
+1. Go to your Jira **Site administration** page. Click on the
+   **dropdown**( <i class="fa fa-ellipsis-h" aria-hidden="true"></i> )
+   under the **Actions** column on your **Jira** app, and select
+   the **Jira Settings** option.
```
I think you could simplify this a bit: "under **Actions** for your **Jira** app, and select **Jira settings**."

From your screenshot of the UI here, it looks like it's "Jira settings" and not "Jira Settings" re:capitalization. Also, should we use "app" here or "product"? I ask because in the screenshot that column is labeled "product".

laurynmm commented on zerver/webhooks/jira/doc.md:
```diff
@@ -14,8 +14,12 @@ Server versions 5.2 or greater. For older versions, you'll need our
+   the **Jira Settings** option.
+
+1. In the left sidebar, scroll down, and under **Advanced**, click
    **WebHooks**. Click **+ Create a WebHook**.
```
I'd combine this with the previous step. It's a bit long, but our general structure for these documentation pages is one step with all navigation instructions then one step with configuration instructions, and repeat as necessary.

laurynmm reviewed (commented):
@PieterCK - The code changes look good to go. I had some feedback on the documentation updates. Let me know if you have any questions.

Also, just looking at your screenshot from the Jira admin UI, if you navigate to the **Settings** page at the very top, is that different from navigating to **Jira settings** via the **Overview** page? If so, could we simplify these instructions to use that navigation?

PieterCK commented:
> Also, just looking at your screenshot from the Jira admin UI, if you navigate to the **Settings** page at the very top, is that different from navigating to **Jira settings** via the **Overview** page? If so, could we simplify these instructions to use that navigation?

Unfortunately it doesn't go to the same page..
![image](https://github.com/user-attachments/assets/aac9c810-1519-4bf7-bafb-b3e68ef25a78)

PieterCK commented:
Updated the doc to address reviews, reformatted the note about compatible Jira version, and updated the doc screenshot in the PR description.

@laurynmm Thanks for the review! Let me know whether the changes have addressed your comments or not

laurynmm commented:
@PieterCK - Thanks for making those updates! I was a bit unclear with one of my comments, so I went ahead and pushed a revision here. See updated documentation screenshot below. Going to go ahead and mark for integration review.

<details>
<summary>Updated Jira integration doc</summary>

[current documentation](https://zulip.com/integrations/doc/jira)
![Screenshot from 2024-08-12 18-44-40](https://github.com/user-attachments/assets/be97957a-a38e-4d15-8a62-029ed9891f53)
</details>

timabbott commented:
Looks great, merged, thanks @PieterCK!

timabbott commented:
This was backported in #31355.
